Unable to build / run npm install / run bower install

Error printed when building / running npm install / running bower install:

> npm install
> ember-data@1.0.0-beta.9 postinstall /home/gaurav/code/emberjs/ember-data
> bower install

bower ember#~1.6.0-beta.5bower ember#~1.6.0-beta.5        error Arguments to path.join must be strings

Stack trace:
TypeError: Arguments to path.join must be strings
    at path.js:360:15
    at Array.filter (native)
    at Object.exports.join (path.js:358:36)
    at GitHubResolver._checkout (/home/gaurav/code/emberjs/ember-data/node_modules/bower/lib/core/resolvers/GitHubResolver.js:54:21)
    at /home/gaurav/code/emberjs/ember-data/node_modules/bower/lib/core/resolvers/GitResolver.js:69:21
    at _fulfilled (/home/gaurav/code/emberjs/ember-data/node_modules/bower/node_modules/q/q.js:798:54)
    at self.promiseDispatch.done (/home/gaurav/code/emberjs/ember-data/node_modules/bower/node_modules/q/q.js:827:30)
    at Promise.promise.promiseDispatch (/home/gaurav/code/emberjs/ember-data/node_modules/bower/node_modules/q/q.js:760:13)
    at /home/gaurav/code/emberjs/ember-data/node_modules/bower/node_modules/q/q.js:574:44
    at flush (/home/gaurav/code/emberjs/ember-data/node_modules/bower/node_modules/q/q.js:108:17)

Console trace:
Trace
    at StandardRenderer.error (/home/gaurav/code/emberjs/ember-data/node_modules/bower/lib/renderers/StandardRenderer.js:74:17)
    at Logger.updateNotifier.packageName (/home/gaurav/code/emberjs/ember-data/node_modules/bower/bin/bower:109:18)
    at Logger.emit (events.js:95:17)
    at Logger.emit (/home/gaurav/code/emberjs/ember-data/node_modules/bower/node_modules/bower-logger/lib/Logger.js:29:39)
    at /home/gaurav/code/emberjs/ember-data/node_modules/bower/lib/commands/install.js:27:16
    at _rejected (/home/gaurav/code/emberjs/ember-data/node_modules/bower/node_modules/q/q.js:808:24)
    at /home/gaurav/code/emberjs/ember-data/node_modules/bower/node_modules/q/q.js:834:30
    at Promise.when (/home/gaurav/code/emberjs/ember-data/node_modules/bower/node_modules/q/q.js:1079:31)
    at Promise.promise.promiseDispatch (/home/gaurav/code/emberjs/ember-data/node_modules/bower/node_modules/q/q.js:752:41)
    at /home/gaurav/code/emberjs/ember-data/node_modules/bower/node_modules/q/q.js:574:44

System info:
Bower version: 1.2.8
Node version: 0.10.29
OS: Linux 3.13.0-32-generic x64

npm ERR! ember-data@1.0.0-beta.9 postinstall: `bower install`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the ember-data@1.0.0-beta.9 postinstall script.
npm ERR! This is most likely a problem with the ember-data package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     bower install
npm ERR! You can get their info via:
npm ERR!     npm owner ls ember-data
npm ERR! There is likely additional logging output above.
npm ERR! System Linux 3.13.0-32-generic
npm ERR! command "/usr/bin/node" "/usr/bin/npm" "install"
npm ERR! cwd /home/gaurav/code/emberjs/ember-data
npm ERR! node -v v0.10.29
npm ERR! npm -v 1.4.14
npm ERR! code ELIFECYCLE
npm ERR! 
npm ERR! Additional logging details can be found in:
npm ERR!     /home/gaurav/code/emberjs/ember-data/npm-debug.log
npm ERR! not ok code 0

Contents of npm-debug.log

0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/node', '/usr/bin/npm', 'install' ]
2 info using npm@1.4.14
3 info using node@v0.10.29
4 verbose node symlink /usr/bin/node
5 verbose readDependencies using package.json deps
6 verbose install where, deps [ '/home/gaurav/code/emberjs/ember-data',
6 verbose install   [ 'aws-sdk',
6 verbose install     'bower',
6 verbose install     'defeatureify',
6 verbose install     'ember-publisher',
6 verbose install     'grunt',
6 verbose install     'grunt-cli',
6 verbose install     'grunt-contrib-clean',
6 verbose install     'grunt-contrib-concat',
6 verbose install     'grunt-contrib-connect',
6 verbose install     'grunt-contrib-copy',
6 verbose install     'grunt-contrib-jshint',
6 verbose install     'grunt-contrib-qunit',
6 verbose install     'grunt-contrib-uglify',
6 verbose install     'grunt-contrib-watch',
6 verbose install     'grunt-contrib-yuidoc',
6 verbose install     'grunt-ember-defeatureify',
6 verbose install     'grunt-es6-module-transpiler',
6 verbose install     'load-grunt-config',
6 verbose install     'load-grunt-tasks',
6 verbose install     'lockfile',
6 verbose install     'matchdep',
6 verbose install     'yuidocjs' ] ]
7 info preinstall ember-data@1.0.0-beta.9
8 warn package.json defeatureify@0.1.4 No description
9 warn package.json defeatureify@0.1.4 No repository field.
10 verbose readDependencies using package.json deps
11 verbose already installed skipping defeatureify@~0.1.4 /home/gaurav/code/emberjs/ember-data
12 verbose already installed skipping aws-sdk@~2.0.0-rc8 /home/gaurav/code/emberjs/ember-data
13 verbose already installed skipping bower@~1.2 /home/gaurav/code/emberjs/ember-data
14 verbose already installed skipping ember-publisher@0.0.3 /home/gaurav/code/emberjs/ember-data
15 verbose already installed skipping grunt-cli@~0.1.13 /home/gaurav/code/emberjs/ember-data
16 verbose already installed skipping grunt@~0.4.2 /home/gaurav/code/emberjs/ember-data
17 verbose already installed skipping grunt-contrib-concat@~0.3.0 /home/gaurav/code/emberjs/ember-data
18 verbose already installed skipping grunt-contrib-clean@~0.5.0 /home/gaurav/code/emberjs/ember-data
19 verbose already installed skipping grunt-contrib-connect@~0.6.0 /home/gaurav/code/emberjs/ember-data
20 verbose already installed skipping grunt-contrib-copy@^0.5.0 /home/gaurav/code/emberjs/ember-data
21 verbose already installed skipping grunt-contrib-qunit@~0.3.0 /home/gaurav/code/emberjs/ember-data
22 verbose already installed skipping grunt-contrib-jshint@~0.8.0 /home/gaurav/code/emberjs/ember-data
23 verbose already installed skipping grunt-contrib-uglify@~0.3.0 /home/gaurav/code/emberjs/ember-data
24 verbose already installed skipping grunt-contrib-watch@~0.5.3 /home/gaurav/code/emberjs/ember-data
25 verbose already installed skipping grunt-ember-defeatureify@~0.1.0 /home/gaurav/code/emberjs/ember-data
26 verbose already installed skipping grunt-contrib-yuidoc@~0.5.0 /home/gaurav/code/emberjs/ember-data
27 verbose already installed skipping grunt-es6-module-transpiler@~0.6.0 /home/gaurav/code/emberjs/ember-data
28 verbose already installed skipping load-grunt-config@~0.7.0 /home/gaurav/code/emberjs/ember-data
29 verbose already installed skipping lockfile@~0.4.2 /home/gaurav/code/emberjs/ember-data
30 verbose already installed skipping load-grunt-tasks@~0.2.1 /home/gaurav/code/emberjs/ember-data
31 verbose already installed skipping matchdep@~0.3.0 /home/gaurav/code/emberjs/ember-data
32 verbose already installed skipping yuidocjs@~0.3.46 /home/gaurav/code/emberjs/ember-data
33 silly resolved []
34 info build /home/gaurav/code/emberjs/ember-data
35 verbose linkStuff [ false, false, false, '/home/gaurav/code/emberjs' ]
36 info linkStuff ember-data@1.0.0-beta.9
37 verbose linkBins ember-data@1.0.0-beta.9
38 verbose linkMans ember-data@1.0.0-beta.9
39 verbose rebuildBundles ember-data@1.0.0-beta.9
40 verbose rebuildBundles [ '.bin',
40 verbose rebuildBundles   'aws-sdk',
40 verbose rebuildBundles   'bower',
40 verbose rebuildBundles   'defeatureify',
40 verbose rebuildBundles   'ember-publisher',
40 verbose rebuildBundles   'grunt',
40 verbose rebuildBundles   'grunt-cli',
40 verbose rebuildBundles   'grunt-contrib-clean',
40 verbose rebuildBundles   'grunt-contrib-concat',
40 verbose rebuildBundles   'grunt-contrib-connect',
40 verbose rebuildBundles   'grunt-contrib-copy',
40 verbose rebuildBundles   'grunt-contrib-jshint',
40 verbose rebuildBundles   'grunt-contrib-qunit',
40 verbose rebuildBundles   'grunt-contrib-uglify',
40 verbose rebuildBundles   'grunt-contrib-watch',
40 verbose rebuildBundles   'grunt-contrib-yuidoc',
40 verbose rebuildBundles   'grunt-ember-defeatureify',
40 verbose rebuildBundles   'grunt-es6-module-transpiler',
40 verbose rebuildBundles   'load-grunt-config',
40 verbose rebuildBundles   'load-grunt-tasks',
40 verbose rebuildBundles   'lockfile',
40 verbose rebuildBundles   'matchdep',
40 verbose rebuildBundles   'yuidocjs' ]
41 info install ember-data@1.0.0-beta.9
42 info postinstall ember-data@1.0.0-beta.9
43 verbose unsafe-perm in lifecycle true
44 info ember-data@1.0.0-beta.9 Failed to exec postinstall script
45 error ember-data@1.0.0-beta.9 postinstall: `bower install`
45 error Exit status 1
46 error Failed at the ember-data@1.0.0-beta.9 postinstall script.
46 error This is most likely a problem with the ember-data package,
46 error not with npm itself.
46 error Tell the author that this fails on your system:
46 error     bower install
46 error You can get their info via:
46 error     npm owner ls ember-data
46 error There is likely additional logging output above.
47 error System Linux 3.13.0-32-generic
48 error command "/usr/bin/node" "/usr/bin/npm" "install"
49 error cwd /home/gaurav/code/emberjs/ember-data
50 error node -v v0.10.29
51 error npm -v 1.4.14
52 error code ELIFECYCLE
53 verbose exit [ 1, true ]

--- a/package.json
+++ b/package.json
@@ -11,7 +11,7 @@
   },
   "devDependencies": {
     "aws-sdk": "~2.0.0-rc8",
-    "bower": "~1.2",
+    "bower": "~1.3",
     "defeatureify": "~0.1.4",
     "ember-publisher": "0.0.3",
     "grunt": "~0.4.2",

